### PR TITLE
[Parser] Parse rec groups

### PR DESCRIPTION
**Pull request by tlively**

### CODE REVIEW

**Comment by tlively** [comment]
Current dependencies on/for this PR:
* main
  * **PR #4784** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/4784" target="_blank">Graphite</a> 
    * **PR #4785** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/4785" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/4785?utm_source=stack-comment).

File: src/wasm/wat-parser.cpp [hunk 1]
```diff
@@ -350,9 +351,13 @@ struct ParseTypeDefsCtx {
   // Parse the names of types and fields as we go.
   std::vector<TypeNames> names;
 
-  // The index of the type definition we are parsing.
+  // The index of the type definition (i.e. rec group) we are parsing. This is
```
Comment (kripken) [inline src/wasm/wat-parser.cpp:354]: Could this be rephrased as "index is the rec group index, typeIndex is the type index inside the rec group"? If so, index => recIndex maybe?

Comment (tlively) [reply src/wasm/wat-parser.cpp]: I stopped using `parseDefs` for type parsing, which let me remove this useless field entirely and rename `typeIndex` to just `index`.

**Review by kripken — APPROVED** [review]

**Comment by tlively** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/4785?utm_source=stack-comment).


Job Reference: wHPoO07O2vDdfBC3TH7I

**Pull request merged**